pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1405,6 +1405,21 @@ def _is_function_def_never_returning(self, node: astroid.FunctionDef) -> bool:
         Returns:
             bool: True if the function never returns, False otherwise.
         """
+        if isinstance(node, astroid.FunctionDef) and node.returns:
+            try:
+                rtype = next(node.returns.infer())
```

Comment:
I'm not sure it's really worth it to infer the `node.returns`. Just checking the `attrname` / `name` for `NoReturn` is probably enough. If it matches, it's quite possibly `typing.NoReturn`. Who would define their own?

Reply:
@cdce8p you are right. It is far more simple to avoid this inference and it is more versatile as it works for all python versions. :+1: